examples/digital_fingerprinting/production/morpheus/dfp/stages/dfp_training.py

```diff
@@ -71,13 +75,18 @@ def on_data(self, message: MultiDFPMessage):
 
         model = AutoEncoder(**self._model_kwargs)
 
-        final_df = message.get_meta_dataframe()
+        train_df = message.get_meta_dataframe()
 
         # Only train on the feature columns
-        final_df = final_df[final_df.columns.intersection(self._config.ae.feature_columns)]
+        train_df = train_df[train_df.columns.intersection(self._config.ae.feature_columns)]
+        validation_df = None
+
+        # Split into training and validation sets
+        if int(self._validation_size * 100) in range(1, 100):
+            train_df, validation_df = train_test_split(train_df, test_size=self._validation_size, shuffle=False)
```

Comment:
This check should be performed in the constructor to make sure the value is between 0.0 and 1.0. At this point, all we should be doing is checking for `self._validation_size > 0.0`